### DISCUSSION TITLE
Show GPX progress via Live Activities

### DISCUSSION THREAD

**Discussion by bryceco** [bryceco/GoMap#751] [Ideas]
I'm playing with adding feedback while recording GPX tracks. On the lock screen it looks something like this:
<img width="344" alt="Screenshot 2024-05-05 at 8 58 04 PM" src="https://github.com/bryceco/GoMap/assets/279734/a8338b0c-1cf2-4428-ab91-bc91ab9f0d93">

In the future we could potentially add buttons for pausing/starting the GPX recording, or other things. Looking for feedback on what would be good to display, and how to style the information.

**Comment by tordans** [comment]
That is an interesting addition. My first thought was, that this is not something I am missing. Probably due to the way I use GoMap: I usually don't have it running without also mapping.

However, this might help solve other issues I have with the GPX feature: I don't want to record right from my start position. And I also don't want to record my end position. The last part is the main problem, because I end up recording for a while and then forget to turn it of. I then have a big pile where I am staying … which I don't want to put on the website. I tend to discard the track or manually edit the file before upload which is quite annoying work.

Those use cases could be solved differently like with a privacy zone that the Mapillary App has where nothing gets recorded. Or some "remove 50 points at the end" button.

But given this feature, maybe a **"split" action** would be a good idea? This way I might have an easier way to split large tracks and discard only the first/last bits?

**Comment by bryceco** [comment]
I'm thinking about this feature because there's a group that primarily want to record GPX tracks and edit them later, rather than mapping as they go. Personally sometimes I map, and sometimes I record and trace later.

@verhovsky mentioned wanting to omit recording traces while at home, which is related to your request. He wants to record essentially 24/7, with exceptions for uninteresting places.

A split action would be something separate, done inside the app itself and not really related to this. I can see highlighting a trace and then having buttons to trim the front/back some number of nodes at a time.

**Reply by tordans** [reply]
> A split action would be something separate, done inside the app itself and not really related to this.

I meant this as a action on the lock screen UI and it would basically be a "stop and start" action which AFAIK would create a separate trace (AKA "split" it).

**Reply by bryceco** [reply]
Yes a stop/start button is the easiest and is equivalent to a split. Some people would prefer a "pause" button which is more complicated.

**Comment by bryceco** [comment]
I wasn't able to make start/stop button on the lock screen. The issue is that if you press "Stop" you don't want the activity to appear as active in the Dynamic Island. But the only way to prevent that is to end the live activity, and if you end it then the Start button doesn't work anymore.
